### PR TITLE
Use Treeherder instead of hg-mo to query for Tinderbox builds (#738)

### PR DESCRIPTION
**Pull request by whimboo**
This is part II for issue #738 which will change the crawling from hg.mo to Treeherder.

### CODE REVIEW

File: lib/automation.py [hunk 1]
```diff
@@ -203,27 +204,26 @@ def get_test_packages_url(self, properties):
...
-
-                    # Stop once we weren't able to retrieve the revision data or
-                    # have a revision not pushed by automation.
-                    if r.status_code != 200 or json_data['user'] != 'ffxbld':
+                revision = properties['revision'][:12]
+
+                client = TreeherderClient(host='treeherder.mozilla.org', protocol='https')
+                resultsets = client.get_resultsets(properties['branch'],
+                                                   tochange=revision,
+                                                   count=50)
```
Comment (whimboo) [inline lib/automation.py]: Usually there are only <5 revisions as made by automation for tagging releases and such. So retrieving 50 to be on the safe side.

**Comment by whimboo** [comment]
@syd, can you please review?

**Comment by Syd** [comment]
I possibly could, but I think you're after @sydvicious :)

**Comment by whimboo** [comment]
Ups. Indeed. Sorry for that.

**Comment by sydvicious** [comment]
@whimboo There seems to be some overlap between this diff and #754 . I assume that this one depends on #754 , and will thus approve it r+.

**Comment by whimboo** [comment]
This one is PR #754. So I assume you mean PR #755. To solve this mystic I have to say that I explicitly said on PR #755 that I did a rebase for this PR which was necessary for testing and to get merge conflicts solved.

**Pull request merged**